scripts/postlink/ios/postlink.js

```diff
@@ -94,3 +98,42 @@ function findFileByAppName(array, appName) {
 
     return null;
 }
+
+function getDefaultPlistPath() {
+    //this is old logic in case we are unable to find PLIST from xcode/pbxproj - at least we can fallback to default solution
+    return glob.sync(`**/${package.name}/*Info.plist`, ignoreNodeModules)[0];
+}
+
+function getPlistPath(){
+    var xcodeProjectPaths = glob.sync(`**/*.xcodeproj/project.pbxproj`, ignoreNodeModules);
+    if (!xcodeProjectPaths){
+        return getDefaultPlistPath();
+    }
+
+    if (xcodeProjectPaths.length !== 1) {
+        console.log('Could not determine correct xcode proj path to retrieve related plist file, there are multiple xcodeproj under the solution.');
+        return getDefaultPlistPath();
+    }
+
+    var xcodeProjectPath = xcodeProjectPaths[0];
+    var parsedXCodeProj;
+
+    try {
+        var proj = xcode.project(xcodeProjectPath);      
+        //use sync version because there are some problems with async version of xcode lib as of current version
+        parsedXCodeProj = proj.parseSync();
+    }
+    catch(e) {
+        console.log('Couldn\'t read info.plist path from xcode project - error: ' + e.message);
+        return getDefaultPlistPath();
+    }
+
+    var INFO_PLIST_PROJECT_KEY = 'INFOPLIST_FILE';
+    var plistPathValue = parsedXCodeProj.getBuildProperty(INFO_PLIST_PROJECT_KEY);
```

Comment:
@vladimir-kotikov, @sergey-akhalkov - I'm tagging you because @max-mironov is out at the moment.

I've looked through this and compared it to the [`project.pbxproj` in our demo app](https://github.com/Microsoft/react-native-code-push/blob/master/Examples/CodePushDemoApp/iOS/CodePushDemoApp.xcodeproj/project.pbxproj). Looking for the `INFOPLIST_FILE` key, I see there are four options depending on project and build:

Debug build for CodePushDemoAppTests
Release build for CodePushDemoAppTests
Debug build for CodePushDemoApp
Release build for CodePushDemoApp

It's unclear to me which `Info.plist` file would actually get used - looking at the [implementation of `getBuildProperty()`](https://github.com/alunny/node-xcode/blob/078625f16c102372965cc5a4f6a8f7a5145d36aa/lib/pbxProject.js#L1952), it seems like it would be the first one found, and you can optionally narrow down by the build type, but not by the project. So, if I understand correctly, it would use the `Info.plist` file for `CodePushDemoAppTest`, which is not what we want, right?